### PR TITLE
Subscription with device event filter

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: service/deviceProjection.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+			var s daEvents.DeviceOffline
+			if err := eu.Unmarshal(&s); err != nil {
+				return err
+			}
+			if m.state.IsOnline {
+				onDeviceOnline = false
+				onDeviceOffline = true
+			}
+			m.state.Id = s.Id
+			m.state.IsOnline = false
```
Comment (cizmazia) [inline service/deviceProjection.go]: It seems that just one of the three `onDeviceOnline`, `onDeviceOffline`, `m.state.IsOnline` should suffice. Do you prevent duplicate notifications?

File: service/resourceProjection.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+		case http.ProtobufContentType(&pbRA.ResourcePublished{}):
+			var s raEvents.ResourcePublished
+			if err := eu.Unmarshal(&s); err != nil {
+				return err
+			}
+			if !m.isPublished {
+				onResourcePublished = true
+				onResourceUnpublished = false
+			}
+			m.isPublished = true
```
Comment (cizmazia) [inline service/resourceProjection.go]: It seems that only one of the three `onResourcePublished`, `onResourceUnpublished`, `m.isPublished` is needed.

File: service/deviceProjection.go (hunk 1 above)
Comment (jkralik) [reply service/deviceProjection.go]: They are needed to resolve of sending events to gprc streams.

File: service/resourceProjection.go (hunk 2 above)
Comment (jkralik) [reply service/resourceProjection.go]: They are needed to resolve of sending events to gprc streams.

**Pull request merged**